### PR TITLE
feat(log-path): log path is defined in the .env

### PR DESCRIPTION
**Pull request by rex9840**

### CODE REVIEW

**Review by PublisherName — COMMENTED** [review]
**Suggestion**: 

- Consider changing `LOG_DIR.mkdir(exist_ok=True)` to `LOG_DIR.mkdir(exist_ok=True, parents=True)`. This adjustment will ensure that any missing parent directories are created, preventing potential FileNotFoundError if the directory path is not fully established.

-  I recommend changing the filename construction from `"filename": f"./{LOG_DIR}/custom.log"` to `"filename": str(LOG_DIR / "custom.log")`. Using the Path object with the / operator is preferable because it ensures proper path handling and platform compatibility. It avoids potential issues with path separators and maintains clarity in the code.

```
LOG_DIR = BASE_DIR / os.getenv("LOG_DIR", "logs")
try:
    LOG_DIR.mkdir(exist_ok=True, parents=True)
    if not os.access(LOG_DIR, os.W_OK):
        raise PermissionError(f"No write permission for log directory: {LOG_DIR}")
except Exception as e:
    raise RuntimeError(f"Failed to setup log directory: {e}")
```

File: core/settings.py [hunk 1]
```diff
@@ -190,7 +190,7 @@
 
 # Logs
 
-LOG_DIR = BASE_DIR / "logs"
+LOG_DIR = BASE_DIR / os.getenv("LOG_PATH", "logs")
 LOG_DIR.mkdir(exist_ok=True)
```
Comment (n1rjal) [inline core/settings.py]: @rex9840  Please conform to the changes requested by @PublisherName as well. Also look into why CI is breaking as well.

**Review by n1rjal — CHANGES_REQUESTED** [review]

File: core/settings.py (hunk 1 above)
Comment (rex9840) [reply core/settings.py]: @n1rjal review as the linting error has been solved

**Comment by rex9840** [comment]
@PublisherName 
BASE_DIR is  django project dir 
and as for the log_dir exitst=true does the validaton for us
as per the expections it only occurs with logs not the dir do we really need it

**Comment by PublisherName** [comment]
@rex9840 If LOG_PATH is set to a restricted folder like `/var/www/logs`, there’s a good chance we’ll run into permission issues. Although Django can catch this, it’s helpful to wrap the setup in a try block. Even better, if a permission error happens, we can fall back to using a safe location like `/tmp/logs` or `BASE_DIR / logs` instead. This way, we don’t need to worry about permission errors stopping the logs from being saved.

```python
default_log_dir = BASE_DIR / "logs"
log_path = os.getenv("LOG_PATH", default_log_dir)
LOG_DIR = Path(log_path)

try:
    LOG_DIR.mkdir(exist_ok=True, parents=True)
    if not os.access(LOG_DIR, os.W_OK):
        raise PermissionError(f"No write permission for log directory: {LOG_DIR}")
except PermissionError as e:
    LOG_DIR = Path("/tmp/logs") # Or Path(default_log_dir)
    LOG_DIR.mkdir(exist_ok=True, parents=True)
    print(f"Warning: Using fallback log directory at {LOG_DIR} due to: {e}")
except Exception as e:
    raise RuntimeError(f"Failed to set up log directory: {e}")
```

**Comment by n1rjal** [comment]
@PublisherName we deploy in platform like open shift or any kubernetes based environment where we don't get enough permissions even to temp folder unless we mount it. So, we will be happy with raising exceptions. 

It's only for openshift environment though your use case works for majority of enivironment.

**Review by n1rjal — APPROVED** [review]

**Review by Sailesh-Singh — APPROVED** [review]

**Pull request merged**